core/domain/exp_domain.py

```diff
@@ -323,41 +323,122 @@ def to_dict(self):
+    def from_dict(cls, audio_translation_dict):
+        return cls(
+            audio_translation_dict['language_code'],
+            audio_translation_dict['filename'],
+            audio_translation_dict['file_size_bytes'],
+            audio_translation_dict['needs_update'])
+
+    def __init__(self, language_code, filename, file_size_bytes, needs_update):
+        # str. A language code, such as ‘en’ or ‘hi’. A hybrid language will be
+        # represented for example as ‘hi-en’ for Hinglish.
+        self.language_code = language_code
+        # str. The corresponding audio file path, e.g.
+        # "content-en-2-h7sjp8s.mp3".
+        self.filename = filename
+        # int. The file size, in bytes. Used to display potential bandwidth
+        # usage to the learner before they download the file.
+        self.file_size_bytes = file_size_bytes
+        # bool. Whether audio is marked for needing review.
+        self.needs_update = needs_update
+
+    def validate(self):
+        if not isinstance(self.language_code, basestring):
+            raise utils.ValidationError(
+                'Invalid language code: %s' % self.language_code)
+        allowed_audio_language_codes = [
+            language['id'] for language in constants.SUPPORTED_AUDIO_LANGUAGES]
+        if self.language_code not in allowed_audio_language_codes:
+            raise utils.ValidationError(
+                'Unrecognized language code: %s' % self.language_code)
 
-    def __init__(self, content_type, value=''):
-        self.type = content_type
-        self.value = html_cleaner.clean(value)
+        if not isinstance(self.filename, basestring):
+            raise utils.ValidationError(
+                'Invalid audio filename: %s' % self.filename)
+        dot_index = self.filename.rfind('.')
+        if dot_index == -1 or dot_index == 0:
+            raise utils.ValidationError(
+                'Invalid audio filename: %s' % self.filename)
+        extension = self.filename[dot_index + 1:]
+        if extension not in feconf.ACCEPTED_AUDIO_EXTENSIONS:
+            raise utils.ValidationError(
+                'Invalid audio filename: it should have one of '
+                'the following extensions: %s. Received: %s',
+                (feconf.ACCEPTED_AUDIO_EXTENSIONS, self.filename))
+
+        if not isinstance(self.file_size_bytes, int):
+            raise utils.ValidationError(
+                'Invalid file size: %s' % self.file_size_bytes)
+        if self.file_size_bytes <= 0:
+            raise utils.ValidationError(
+                'Invalid file size: %s' % self.file_size_bytes)
+
+        if not isinstance(self.needs_update, bool):
+            raise utils.ValidationError(
+                'Invalid needs_update attribute: %s' % self.needs_update)
+
+
+class SubtitledHtml(object):
+    """Value object representing subtitled HTML."""
+
+    def to_dict(self):
+        return {
+            'html': self.html,
+            'audio_translations': [
```

Comment:
I think AudioTranslation might be better, because it explicitly conveys translation into other languages, whereas AudioSubtitle doesn't. So I lean towards leaving it as-is.